app/lib/nomis_client/base.rb

```diff
@@ -69,6 +79,41 @@ def update_json_headers(params)
           },
         }.deep_merge(params)
       end
+
+      def paginate_through(path, params = {}, verb = :get)
+        responses = []
+
+        params = DEFAULT_PAGINATE_PARAMS.dup.deep_merge(params)
+        response = NomisClient::Base.send(verb, path, params)
+        responses << response
+        total_records = response.headers['Total-Records'].to_i
+        found_records = 0
+
+        return [] if total_records < 1
+
+        if response.status == 200
+          collection_items = yield response.parsed
+
+          found_records += collection_items.count
+        end
+
+        while found_records < total_records
+          offset = params[:headers]['Page-Offset'].to_i
+          params[:headers]['Page-Offset'] = (offset + 1).to_s
+
+          response = NomisClient::Base.send(verb, path, params)
+          responses << response
+          total_records = response.headers['Total-Records'].to_i
+
+          if response.status == 200
+            collection_items = yield response.parsed
+
+            found_records += collection_items.count
+          end
+        end
+
+        responses
+      end
```

Comment:
it looks a bit complex, it would be nice to simplify it, maybe extracting part of the functionality in another private method